src/theme/NavbarItem/DropdownNavbarItem.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import DropdownNavbarItem from "@theme-original/NavbarItem/DropdownNavbarItem";
+
+export default function DropdownNavbarItemWrapper({ label, items, ...props }) {
+  if (label === "Next") {
+    label = "8.7 (unreleased)";
+  }
+
+  items = items.map((item) => {
+    let { label } = item;
+    if (label === "Next") {
+      label = "8.7 (unreleased)";
+    }
```

Comment:
We'd likely make this text a configuration setting, since it will probably be used in additional locations.